### PR TITLE
Fix issues with pings and doppelgänger detection.

### PR DESCRIPTION
**Pull request by afck**
Closes https://github.com/casper-network/Governance/issues/36.

### CODE REVIEW

File: node/src/components/consensus/highway_core/state.rs [hunk 1]
```diff
@@ -568,13 +568,11 @@ impl<C: Context> State<C> {
         self.pings[creator] = self.pings[creator].max(timestamp);
     }
 
-    /// Returns `true` if the latest timestamp we have is less than one maximum round length older
-    /// than the given timestamp.
-    ///
-    /// This is to prevent ping spam: If the incoming ping is only slightly newer than a unit or
-    /// ping we have already received, we drop it without forwarding it to our peers.
+    /// Returns `true` if the latest timestamp we have is older than the given timestamp.
```
Comment (xcthulhu) [inline node/src/components/consensus/highway_core/state.rs:571]: I'm trying to understand.

If the ping was not created by us, it seems the old behavior makes sense.

And if the ping was from us, perhaps we could prevent ourselves from forwarding it elsewhere?

Comment (xcthulhu) [reply node/src/components/consensus/highway_core/state.rs:571]: Looks like we aren't forwarding, so this is the correct behavior.

**Review by xcthulhu — APPROVED** [review]
LGTM

File: node/src/components/consensus/highway_core/highway.rs [hunk 2]
```diff
@@ -219,7 +219,7 @@ impl<C: Context> Highway<C> {
             self.instance_id,
         );
         self.active_validator = Some(av);
-        effects
+        self.add_new_own_vertices(effects, current_time)
```
Comment (goral09) [inline node/src/components/consensus/highway_core/highway.rs:222]: That's the root of the problem? We were not adding our own ping to the state?

**Review by goral09 — APPROVED** [review]

File: node/src/components/consensus/highway_core/highway.rs (hunk 2 above)
Comment (afck) [reply node/src/components/consensus/highway_core/highway.rs:222]: Yes, exactly.

And the other potential problem was that we weren't counting pings from someone else as doppelgänger pings if they weren't _much_ younger (more than one maximum round length) than the latest one we had in the state.

Comment (goral09) [reply node/src/components/consensus/highway_core/highway.rs:222]: Thanks 👍

**Review by goral09 — APPROVED** [review]

**Comment by afck** [comment]
bors merge

**Pull request merged**